pkg/manifest/jsonmanifest/manifest.go

```diff
@@ -6,51 +6,94 @@ package jsonmanifest
 
 import (
 	"encoding/json"
+	"sync"
 
 	"github.com/ethersphere/bee/pkg/manifest"
 )
 
-// verify JSONManifest implements manifest.Interface.
-var _ manifest.Interface = (*JSONManifest)(nil)
+// verify jsonManifest implements manifest.Interface.
+var _ manifest.Interface = (*jsonManifest)(nil)
 
-// JSONManifest is a JSON representation of a manifest.
+// jsonManifest is a JSON representation of a manifest.
 // It stores manifest entries in a map based on string keys.
-type JSONManifest struct {
-	Entries map[string]*JSONEntry `json:"entries,omitempty"`
+type jsonManifest struct {
+	entriesMu sync.RWMutex // mutex for accessing the entries map
+	entries   map[string]*jsonEntry
 }
 
-// NewManifest creates a new JSONManifest struct and returns a pointer to it.
-func NewManifest() *JSONManifest {
-	return &JSONManifest{
-		Entries: make(map[string]*JSONEntry),
+// NewManifest creates a new jsonManifest struct and returns a pointer to it.
+func NewManifest() manifest.Interface {
+	return &jsonManifest{
+		entries: make(map[string]*jsonEntry),
 	}
 }
 
 // Add adds a manifest entry to the specified path.
-func (m *JSONManifest) Add(path string, entry manifest.Entry) {
-	m.Entries[path] = NewEntry(entry.Reference(), entry.Name(), entry.Headers())
+func (m *jsonManifest) Add(path string, entry manifest.Entry) {
+	m.entriesMu.Lock()
+	defer m.entriesMu.Unlock()
+
+	m.entries[path] = &jsonEntry{
```

Comment:
Use `NewEntry`?

Reply:
since `NewEntry` returns a `manifest.Entry` type, and `entries` is of the `map[string]*jsonEntry` type, using `NewEntry` here would require a type assertion.

i don't think it's worth it in this case since this is internal to the package, and—for now—we are not handling errors in the `Add` function anyways (i.e. what if the assertion fails?)

Reply:
Yes, I see, it is fine like this.